posthog/schema_helpers.py

```diff
@@ -0,0 +1,81 @@
+from pydantic import BaseModel
+
+from posthog.schema import (
+    FunnelStepReference,
+    FunnelsFilter,
+    FunnelsQuery,
+    FunnelVizType,
+    FunnelConversionWindowTimeUnit,
+    FunnelLayout,
+    BreakdownAttributionType,
+    StepOrderValue,
+)
+
+
+def clean_funnels_filter(funnelsFilter: FunnelsFilter | None) -> FunnelsFilter:
```

Comment:
I think this would be less dreadful if we implemented the cleaning at `@model_serializer` level. See: https://github.com/pydantic/pydantic/discussions/9108#discussioncomment-8919938